localstack-core/localstack/services/ec2/patches.py

```diff
@@ -92,12 +127,21 @@ def ec2_create_vpc(
         custom_ids = [tag["Value"] for tag in tags if tag["Key"] == TAG_KEY_CUSTOM_ID]
         custom_id = custom_ids[0] if len(custom_ids) > 0 else None
 
+        # try to generate a custom id from the id store
+        if not custom_id:
+            resource_identifier = VpcIdentifier(self.account_id, self.region_name, cidr_block)
+            id = localstack_id_manager.get_custom_id(resource_identifier)
+            if id:
+                custom_id = id
+
         # Check if custom id is unique
         if custom_id and custom_id in self.vpcs:
             raise InvalidVpcDuplicateCustomIdError(custom_id)
```

Comment:
Using the generate method will do all the work for you! It also looks at the `TAG_KEY_CUSTOM_ID`, and duplicate ids. It will not raise though :thinking: and instead will generate a new id... so maybe you will want to leave that part out?

```suggestion
        resource_identifier = VpcIdentifier(self.account_id, self.region_name, cidr_block)
        custom_id = resource_identifier.generate(tags=tags, existing_ids=self.vpcs.keys())
```